src/theory/sep/theory_sep.cpp

```diff
@@ -1825,7 +1825,21 @@ bool TheorySep::checkPto(HeapAssertInfo* e, Node p, bool polarity)
       else if (polarity != pol)
       {
         // a positive and negative pto
-        if (!areDisequal(pval, qval))
+        bool isSat = false;
+        std::vector<Node> conc;
+        for (size_t j = 0; j < 2; j++)
```

Comment:
It was to hard to understand this loop without the documentation for the lemma below. 
Perhaps it would be better to document this as well.